crates/bevy_pbr/src/light.rs

```diff
@@ -256,9 +227,249 @@ pub struct DirectionalLightShadowMap {
 impl Default for DirectionalLightShadowMap {
     fn default() -> Self {
         #[cfg(feature = "webgl")]
-        return Self { size: 2048 };
+        return Self { size: 1024 };
         #[cfg(not(feature = "webgl"))]
-        return Self { size: 4096 };
+        return Self { size: 2048 };
+    }
+}
+
+/// Controls how cascaded shadow mapping works.
+#[derive(Component, Clone, Debug, Reflect)]
+#[reflect(Component)]
+pub struct CascadeShadowConfig {
+    /// The (positive) distance to the far boundary of each cascade.
+    pub bounds: Vec<f32>,
+    /// The proportion of overlap each cascade has with the previous cascade.
+    pub overlap_proportion: f32,
+}
+
+impl Default for CascadeShadowConfig {
+    fn default() -> Self {
+        if cfg!(feature = "webgl") {
+            // Currently only support one cascade in webgl.
+            Self::new(1, 5.0, 100.0, 0.2)
+        } else {
+            Self::new(4, 5.0, 1000.0, 0.2)
+        }
+    }
+}
+
+fn calculate_cascade_bounds(
+    num_cascades: usize,
+    nearest_bound: f32,
+    shadow_maximum_distance: f32,
+) -> Vec<f32> {
+    if num_cascades == 1 {
+        return vec![shadow_maximum_distance];
+    }
+    let base = (shadow_maximum_distance / nearest_bound).powf(1.0 / (num_cascades - 1) as f32);
+    (0..num_cascades)
+        .map(|i| nearest_bound * base.powf(i as f32))
+        .collect()
+}
+
+impl CascadeShadowConfig {
+    /// Returns a cascade config for `num_cascades` cascades, with the first cascade
+    /// having far bound `nearest_bound` and the last cascade having far bound `shadow_maximum_distance`.
+    /// In-between cascades will be exponentially spaced.
+    pub fn new(
+        num_cascades: usize,
+        nearest_bound: f32,
+        shadow_maximum_distance: f32,
+        overlap_proportion: f32,
+    ) -> Self {
+        assert!(num_cascades > 0, "num_cascades must be positive");
```

Comment:
~~`num_cascades` is an size, it can't be negative~~
oh that's actually a cultural thing, in French 0 is positive but not in English